backends/s3/s3.go

```diff
@@ -155,6 +155,12 @@ func (b *Backend) doList(ctx context.Context, prefix string) (simpleblob.BlobLis
 		Recursive: false,
 	})
 	for obj := range objCh {
+		// Handle error returned by MinIO client
+		if err := convertMinioError(obj.Err); err != nil {
```

Comment:
I have a feeling this needs to explicitly handle `os.ErrDoesNotExist` and just return an empty list in that case, but I am not sure if this is the desired behaviour.

How do various S3 API implementations handle the situation where you created a bucket and try to list a sub path when you have not created any objects yet? There is no explicit mkdir.

Reply:
It would be useful to add this scenario to `tester.DoBackendTests` to enforce a certain expected behaviour.

Reply:
Yep, will add tests to ensure the behaviour is correct, including the scenario where a prefix is provided which matches 0 entries